Add prometheus exporter to openshift configuration

KalmanMeth opened the pull request:
Includes yaml to run the skydive-prometheus-connector on openshift.
So far, we have only the connector, but it does not perform the necessary configurations on the prometheus side, which must still be done manually.
@eranra

eranra commented:
@KalmanMeth I did not deploy just looked on the .yml ... this looks good to me .... I suggest::

(1) Either add another container with local prometious and connect it to this deployment of add needed objects to connect to OCP prometheous 
(2) Add documentation in readme that explain how to know that this is working and see the metrics in prometheous

KalmanMeth commented:
I added another container with a local prometheus and connected it to this deployment. I also updated the README with instructions to install it and verify that it is working.

eranra commented on contrib/openshift/prometheus-connector-template.yaml:
```diff
@@ -0,0 +1,180 @@
+#
+# Containers included in this pod::
+# (1) Skydive Prometheus Connector - Flow exporter (connects to Skydive Analyzer)
+# (2) Prometheus
```
Prometheus is deployed as anoter pod ... please change the remark to indicate that it is not in this pod ... but deployed in a second pod

KalmanMeth replied:
OK

eranra commented:
@KalmanMeth can you add needed steps to deploy prometious connector to the file `deploy_skydive_ocp.sh` so that it will be easy to deploy with one click. THX

eranra commented on contrib/openshift/README.md:
```diff
@@ -63,6 +71,14 @@ Perform the Following steps to access skydive UI
+ - Point your browser to prometheus end-point  
+ `oc get route prometheus -o jsonpath='http://{.spec.host}'`
+ - Go to the popdown menu Status/Targets
+ - Verify that you see the http://skydive-prometheus-connector:9100/metrics endpoint
```
This worked for me and I was able to see in the UI that `http://skydive-prometheus-connector:9100/metrics` exists and the state is "UP"

eranra commented on contrib/openshift/README.md:
```diff
@@ -63,6 +71,14 @@ Perform the Following steps to access skydive UI
+ - Go to the popdown menu Status/Targets
+ - Verify that you see the http://skydive-prometheus-connector:9100/metrics endpoint
+ - Go back to the home page
+ - In the query bar, write the variable `skydive_network_connection_total_bytes` and Execute
```
@KalmanMeth I chould not see metrics (e.g. skydive_network_connection_total_bytes)  ... looks like something is not workking as you expect:

The logs of the pod looks ok, I see connection ...  but there are no metrics

```2021-03-02T13:33:34.749Z        INFO    core/classify.go:103 NewClassifySubnetFromList  : Cluster net masks: []
2021-03-02T13:33:34.749Z        INFO    pkg/store_prom.go:221 NewStorePrometheus        : prometheus skydive port = 9100
2021-03-02T13:33:34.749Z        INFO    pkg/store_prom.go:226 NewStorePrometheus        : connection timeout = 60
2021-03-02T13:33:34.749Z        INFO    core/subscriber.go:59 NewSubscriber     : Subscribing to ws://skydive-analyzer:8082/ws/subscriber/flow with namespace 'flow'
2021-03-02T13:33:34.750Z        INFO    websocket/client.go:527 (*Client).Connect       : Connecting to ws://skydive-analyzer:8082/ws/subscriber/flow
2021-03-02T13:33:34.767Z        INFO    websocket/client.go:561 (*Client).Connect       : Connected to ws://skydive-analyzer:8082/ws/subscriber/flow
```

Can you re-check ???

KalmanMeth replied:
If there is no capture, then nothing is reported. Verify that we are performing a capture on some valid network interface in your setup.

eranra replied:
All good now ... I am removing this comment

eranra commented:
/lgtm

KalmanMeth commented:
The prometheus exporter functionality was included in the skydive-golang-operator. https://github.com/skydive-project/skydive-golang-operator
